Source/Dafny/Plugins/Configuration.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Dafny.Plugins;
+
+/// <summary>
+/// Plugins should define a class that extends Configuration,
+/// in order to receive plugin-specific command-line arguments by overwriting the method `ParseArguments`
+/// It is also used to provide to Dafny a list of Rewriter using the method `GetRewriters`
+///
+/// If the plugin defines no Configuration, then Dafny will instantiate every sub-class
+/// of Rewriter from the plugin, providing them with an ErrorReporter in the constructor
+/// as the first and only argument.
+/// </summary>
+public abstract class Configuration {
```

Comment:
Consider naming this `PluginConfiguration`, even though it's already in the `Plugins` namespace